### DISCUSSION TITLE
How to migrate data from the backend storage (cassandra)

### DISCUSSION THREAD

**Discussion by vitaliishandra** [JanusGraph/janusgraph#3498] [Q&A] [answered]
Currently we use Janusgraph+Cassandra. Now we are going to migrate to the ScyllaDB. 
We migrated the whole janusgraph keyspace from Cassandra to ScyllaDB and point Janusgraph to cassandra. For migation we used COPY FROM/TO commands.
As a result some queries are failing:
g.V().has('Account', 1) - works greate (returns 18380 vertexes)
g.V().has('Account', 1) - returns NullPointerException
mgmt = graph.openManagement(); mgmt.printSchema(); - returns Could not find type for id: 103437.

Does it mean that we broke some data during the migration? Do we need to migrate any other keyspace?
Soure: Cassandra [cqlsh 5.0.1 | Cassandra 3.11.10 | CQL spec 3.4.4 | Native protocol v4]
Destination: ScyllaDB Cassandra version: [cqlsh 5.0.1 | Cassandra 3.0.8 | CQL spec 3.3.1 | Native protocol v4]

**Comment by porunov** [comment, marked as answer]
Most probably you either didn't migrate data correctly or there are some problems with your cluster. Try to use Repair on ScyllaDB nodes and see if that helps.
https://docs.scylladb.com/stable/operating-scylla/procedures/maintenance/repair.html

ScyllaDB also has docs about migration: https://docs.scylladb.com/stable/using-scylla/migrate-scylla.html

**Reply by porunov** [reply]
P.S. Notice, that Repair has to be triggered on each inconsistent node (probably it's safer to trigger it on all nodes in your case).

**Comment by vitaliishandra** [comment]
@porunov thank you, we migrated data one more time and everything is ok now.
